### PR TITLE
controlling generation docs

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/controlling_generation.mdx:1]:
```suggestion
<!--Copyright 2023 The HuggingFace Team. All rights reserved.
```

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+Controlling outputs generated by diffusion models is an active research topic. In many popular diffusion models, subtle changes in inputs, both images and text prompts, can drastically change outputs. In an ideal world we want to be able to control how semantics are preserved and changed. 
+
+Most examples of preserving semantics reduce to being able to accurately map a change in input to a change in output. I.e. adding an adjective to a subject in a prompt preserves the entire image, only modifying the changed subject. Or, image variation of a particular subject preserves the subject's pose.
+
+Additionally, there are qualities of generated images that we would like to influence beyond semantic preservation. I.e. in general, we would like our outputs to be of good quality, adhere to a particular style, or be realistic.
+
+We will document some of how diffusers supports controlling generation of diffusion models. Much is cutting edge research and can be quite nuanced. If something needs clarifying or you have a suggestion, don't hesitate to open a discussion on the [forum](https://discuss.huggingface.co/) or open an [issue](https://github.com/huggingface/diffusers/issues).
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/controlling_generation.mdx:22]:
```suggestion
In this document, we discuss how `diffusers` supports controlling generation of diffusion models. Much is cutting edge research and can be quite nuanced. If something needs clarifying or you have a suggestion, don't hesitate to open a discussion on the [forum](https://discuss.huggingface.co/) or open an [issue](https://github.com/huggingface/diffusers/issues).
```

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Instruct pix2pix
+
+[Instruct pix2pix](../api/pipelines/stable_diffusion/pix2pix) is fine-tuned from stable diffusion to support editing input images. It takes as input an image with a prompt describing an edit, and it outputs the edited image.
+
+
+## depth2image
+
+[Depth2image](../pipelines/stable_diffusion/depth2img) is fine-tuned from stable diffusion to better preserve semantics for text guided image variation. 
+
+It conditions on a monocular depth estimate of the original image.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/controlling_generation.mdx:66]:
```suggestion
It conditions on a monocular depth estimate of the original image.

The above-mentioned techniques don't essentially fine-tune any of the sub-models, i.e., VAE, UNet, and the text encoder. 

One can also use techniques like [DreamBooth](https://huggingface.co/docs/diffusers/main/en/training/dreambooth) or [Textual Inversion](https://huggingface.co/docs/diffusers/main/en/training/text_inversion) to have personalized control over the generated outputs. But note that these techniques require additional fine-tuning.  

Depending on the use case, one should choose a technique accordingly. In many cases, these techniques could be combined. For example, one could combine Textual Inversion with SEGA to provide more semantic guidance to the outputs generated using Textual Inversion.
```

**Review by sayakpaul — COMMENTED** [review]
Thanks for working on this! Looks pretty good already.

File: docs/source/en/_toctree.yml [hunk 4]
```diff
@@ -32,6 +32,8 @@
       title: Text-Guided Image-Inpainting
     - local: using-diffusers/depth2img
       title: Text-Guided Depth-to-Image
+    - local: using-diffusers/controlling_generation
+      title: Controlling generation
```
Comment (pcuenca) [inline docs/source/en/_toctree.yml:36]: Just a note: we have a mixture of title-casing and sentence-casing all over the place, for example `Text-Guided Depth-to-Image` but `Reusing seeds for deterministic generation`. Same thing for the headings inside the docs. A very minor thing, but at some point we need to standardize on something (/cc @yiyixuxu).

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+
+# Controlling generation of diffusion models
+
+Controlling outputs generated by diffusion models is an active research topic. In many popular diffusion models, subtle changes in inputs, both images and text prompts, can drastically change outputs. In an ideal world we want to be able to control how semantics are preserved and changed.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/controlling_generation.mdx:16]:
```suggestion
Controlling outputs generated by diffusion models has been long pursued by the community and is now an active research topic. In many popular diffusion models, subtle changes in inputs, both images and text prompts, can drastically change outputs. In an ideal world we want to be able to control how semantics are preserved and changed. 
```

Maybe a shoutout to the community as well? (many of these research techniques are inspired by previous work in the community)

File: docs/source/en/using-diffusers/controlling_generation.mdx (hunk 2 above)
Comment (pcuenca) [inline docs/source/en/using-diffusers/controlling_generation.mdx:22]:
```suggestion
We will document some of the techniques `diffusers` supports to control generation of diffusion models. Much is cutting edge research and can be quite nuanced. If something needs clarifying or you have a suggestion, don't hesitate to open a discussion on the [forum](https://discuss.huggingface.co/) or a [GitHub issue](https://github.com/huggingface/diffusers/issues).
```

(Found the first sentence hard to parse, it's probably a "just-me" thing)

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+We provide a high level explanation of how the generation can be controlled as well as a snippet of the technicals. For more in depth explanations on the technicals, the original papers which are linked from the pipelines are always the best resources.
+
+Unless otherwise mentioned, these are techniques that work with existing models and don't require their own weights.
+
+
+## Pix2PixZero
+
+[Pix2Pix-zero](../api/pipelines/stable_diffusion/pix2pix_zero) allows modifying an image from one concept to another while preserving general image semantics.
+
+The denoising process is guided from one conceptual embedding towards another conceptual embedding. The intermediate latents are optimized during the denoising process to push the attention maps towards reference attention maps. The reference attention maps are from the denoising process of the input image and are used to encourage semantic preservation.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/controlling_generation.mdx:33]: This requires gradients then? (Sorry, didn't have a chance to read the code yet). If there are consequences in terms of memory or inference time it'd be cool to briefly mention them :)

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+[Attend and excite](../api/pipelines/stable_diffusion/attend_and_excite) allows subjects in the prompt to be faithfully represented in the final image. 
+
+A set of token indices are given as input. During denoising, each token index is insured to have above a minimum attention threshold for at least one patch of the image. The intermediate latents are iteratively optimized during the denoising process to strengthen the attention of the most neglected subject token until the attention threshold is passed for all subject tokens.
+
+
+## SEGA - Semantic guidance
+
+SEGA allows applying or removing one or more concepts from an image. The strength of the concept can also be controlled. I.e. the smile concept can be used to incrementally increase or decrease the smile of a portrait.
+
+Similar to how classifier free guidance provides guidance via empty prompt inputs, SEGA provides guidance on conceptual prompts. Multiple of these conceptual prompts can be applied simultaneously. Each conceptual prompt can either add or remove their concept depending on if the guidance is applied positively or negatively.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/controlling_generation.mdx:47]: Maybe link to some section about classifier-free guidance, or the paper? I found this in our docs, not sure if we have something else: https://huggingface.co/docs/diffusers/api/pipelines/stable_diffusion/text2img#diffusers.StableDiffusionPipeline.__call__.guidance_scale

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+## SAG - Self attention guidance
+
+[Self attention guidance](../api/pipelines/stable_diffusion/self_attention_guidance) improves the general quality of images.
+
+SAG provides guidance from predictions not conditioned on high frequency details to fully conditioned images. The high frequency details are extracted out of the UNet self-attention maps.
+
+
+## Instruct pix2pix
+
+[Instruct pix2pix](../api/pipelines/stable_diffusion/pix2pix) is fine-tuned from stable diffusion to support editing input images. It takes as input an image with a prompt describing an edit, and it outputs the edited image.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/controlling_generation.mdx:59]: Maybe mention they used GPT-3 and Stable Diffusion to generate the training data and found it generalizes to real image instruction prompts :)

**Review by pcuenca — APPROVED** [review]
Great overview! It'd be cool if we could provide some, errr, guidance, on when we should try one method vs another (speed, memory, quality). This is probably not straightforward, so we can do it later.

File: docs/source/en/using-diffusers/controlling_generation.mdx (hunk 5 above)
Comment (patil-suraj) [reply docs/source/en/using-diffusers/controlling_generation.mdx:16]: +1

File: docs/source/en/using-diffusers/controlling_generation.mdx (hunk 7 above)
Comment (patil-suraj) [inline docs/source/en/using-diffusers/controlling_generation.mdx:47]: Think we should also add links to the papers/code here.

File: docs/source/en/using-diffusers/controlling_generation.mdx (hunk 8 above)
Comment (patil-suraj) [inline docs/source/en/using-diffusers/controlling_generation.mdx:59]: Here we should mention that this requires it's own weights, as in the doc above we wrote

Unless otherwise mentioned, these are techniques that work with existing models and don't require their own weights.

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## depth2image
+
+[Depth2image](../pipelines/stable_diffusion/depth2img) is fine-tuned from stable diffusion to better preserve semantics for text guided image variation.
```
Comment (patil-suraj) [inline docs/source/en/using-diffusers/controlling_generation.mdx:64]: same comment as above.

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for documenting this!

File: docs/source/en/using-diffusers/controlling_generation.mdx (hunk 1 above)
Comment (patrickvonplaten) [reply docs/source/en/using-diffusers/controlling_generation.mdx:1]: It's fine we can update those all at once :-)

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+[Pix2Pix-zero](../api/pipelines/stable_diffusion/pix2pix_zero) allows modifying an image from one concept to another while preserving general image semantics.
+
+The denoising process is guided from one conceptual embedding towards another conceptual embedding. The intermediate latents are optimized during the denoising process to push the attention maps towards reference attention maps. The reference attention maps are from the denoising process of the input image and are used to encourage semantic preservation.
+
+
+## Attend and excite
+
+[Attend and excite](../api/pipelines/stable_diffusion/attend_and_excite) allows subjects in the prompt to be faithfully represented in the final image. 
+
+A set of token indices are given as input. During denoising, each token index is insured to have above a minimum attention threshold for at least one patch of the image. The intermediate latents are iteratively optimized during the denoising process to strengthen the attention of the most neglected subject token until the attention threshold is passed for all subject tokens.
```
Comment (yiyixuxu) [inline docs/source/en/using-diffusers/controlling_generation.mdx:40]: A set of token indices are given as input, corresponding to the subjects in the prompt that need to be present in the image.

Comment (yiyixuxu) [inline docs/source/en/using-diffusers/controlling_generation.mdx:40]:
```suggestion
A set of token indices are given as input, corresponding to the subjects in the prompt that need to be present in the image. During denoising, each token index is insured to have above a minimum attention threshold for at least one patch of the image. The intermediate latents are iteratively optimized during the denoising process to strengthen the attention of the most neglected subject token until the attention threshold is passed for all subject tokens.
```

File: docs/source/en/using-diffusers/controlling_generation.mdx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+Unless otherwise mentioned, these are techniques that work with existing models and don't require their own weights.
+
+
+## Pix2PixZero
+
+[Pix2Pix-zero](../api/pipelines/stable_diffusion/pix2pix_zero) allows modifying an image from one concept to another while preserving general image semantics.
+
+The denoising process is guided from one conceptual embedding towards another conceptual embedding. The intermediate latents are optimized during the denoising process to push the attention maps towards reference attention maps. The reference attention maps are from the denoising process of the input image and are used to encourage semantic preservation.
+
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/controlling_generation.mdx:34]: @omerbt may be you could just add a suggestion for a section on Panorama? Like below

```suggestion

## MultiDiffusion Panorama

...
```
